IdentityCore/tests/MSIDCommonCredentialTests.m

```diff
@@ -0,0 +1,152 @@
+
+@end
+
+@implementation MSIDCommonCredentialTests
+
+#pragma mark - setUp / tearDown
+
+- (void)setUp {
+    [super setUp];
+    // Put setup code here. This method is called before the invocation of each
+    // test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each
+    // test method in the class.
+    [super tearDown];
+}
+
+#pragma mark - MSIDCommonCredential tests
+
+- (void)testcredentialTypeAsString_whenAccessTokenType_shouldReturnAccessTokenString {
+    NSString *result = [MSIDCredentialTypeHelpers credentialTypeAsString:MSIDAccessTokenType];
+    XCTAssertEqualObjects(result, @"AccessToken");
+}
+
+- (void)testcredentialTypeAsString_whenRefreshTokenType_shouldReturnRefreshTokenString {
+    NSString *result = [MSIDCredentialTypeHelpers credentialTypeAsString:MSIDRefreshTokenType];
+    XCTAssertEqualObjects(result, @"RefreshToken");
+}
+
+- (void)testcredentialTypeAsString_whenIDTokenType_shouldReturnIDTokenString {
+    NSString *result = [MSIDCredentialTypeHelpers credentialTypeAsString:MSIDIDTokenType];
+    XCTAssertEqualObjects(result, @"IdToken");
+}
+
+- (void)testcredentialTypeAsString_whenLegacyTokenType_shouldReturnLegacyTokenString {
+    NSString *result = [MSIDCredentialTypeHelpers credentialTypeAsString:MSIDLegacySingleResourceTokenType];
+    XCTAssertEqualObjects(result, @"LegacySingleResourceToken");
+}
+
+- (void)testcredentialTypeAsString_whenOtherTokenType_shouldReturnOtherTokenString {
+    NSString *result = [MSIDCredentialTypeHelpers credentialTypeAsString:MSIDCredentialTypeOther];
+    XCTAssertEqualObjects(result, @"token");
+}
+
+- (void)testTokenTypeFromString_whenAccessTokenString_shouldReturnAccessTokenType {
+    MSIDCredentialType result = [MSIDCredentialTypeHelpers credentialTypeFromString:@"AccessToken"];
+    XCTAssertEqual(result, MSIDAccessTokenType);
+}
+
+- (void)testTokenTypeFromString_whenRefreshTokenString_shouldReturnRefreshTokenType {
+    MSIDCredentialType result = [MSIDCredentialTypeHelpers credentialTypeFromString:@"RefreshToken"];
+    XCTAssertEqual(result, MSIDRefreshTokenType);
+}
+
+- (void)testTokenTypeFromString_whenIDTokenString_shouldReturnIDTokenType {
+    MSIDCredentialType result = [MSIDCredentialTypeHelpers credentialTypeFromString:@"IdToken"];
+    XCTAssertEqual(result, MSIDIDTokenType);
+}
+
+- (void)testTokenTypeFromString_whenLegacyTokenString_shouldReturnLegacyTokenType {
+    MSIDCredentialType result = [MSIDCredentialTypeHelpers credentialTypeFromString:@"LegacySingleResourceToken"];
+    XCTAssertEqual(result, MSIDLegacySingleResourceTokenType);
+}
+
+- (void)testTokenTypeFromString_whenOtherTokenString_shouldReturnOtherTokenType {
+    MSIDCredentialType result = [MSIDCredentialTypeHelpers credentialTypeFromString:@"token"];
+    XCTAssertEqual(result, MSIDCredentialTypeOther);
+}
+
+#pragma mark - IsEqualToItem handling
+
+- (void)testCredentialIsEqualToItemBehavior {
```

Comment:
Looks like we can split it to 2 tests: 1) json parsing 2) isEqual test. Btw, you can find json parsing related tests at MSIDCredentialCacheItemTests.

Reply:
Please also follow the tests naming convention: test[Method]_when[Condition]_should[Result]. Also, no need to duplicate tests (please see my note on code duplication here).